go_lib/bassa.go

```diff
@@ -0,0 +1,738 @@
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+	return string(out)
+}
+
+// BlockUserRequest : Function to block user request
+func (b *Bassa) BlockUserRequest(userName string) {
+	if userName == "" {
+		panic(errIncompleteParams)
+	}
+	endpoint := "/api/user/blocked"
+	apiURL := b.apiURL + endpoint + "/" + userName
+
+	request, err := http.NewRequest("POST", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+}
+
+// UnBlockUserRequest : Function to unblock user request
+func (b *Bassa) UnBlockUserRequest(userName string) {
+	if userName == "" {
+		panic(errIncompleteParams)
+	}
+	endpoint := "/api/user/blocked"
+	apiURL := b.apiURL + endpoint + "/" + userName
+
+	request, err := http.NewRequest("DELETE", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+}
+
+// GetDownloadUserRequests : Function to get download user requests
+func (b *Bassa) GetDownloadUserRequests(limit int) string {
+	if limit == 0 {
+		limit = 1
+	}
+	endpoint := "/api/user/downloads"
+	apiURL := b.apiURL + endpoint + "/" + string(limit)
+
+	request, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+	return string(out)
+}
+
+// GetToptenHeaviestUsers : Function to get top ten heaviest users
+func (b *Bassa) GetToptenHeaviestUsers() string {
+
+	endpoint := "/api/user/heavy"
+	apiURL := b.apiURL + endpoint
+
+	request, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
```

Comment:
Maybe use logger to log these? https://golang.org/pkg/log/

Reply:
Will update with logger